### PR TITLE
Refactor/use latest requests

### LINKED ISSUE #721: Chrome Extension: Use latest react-store

### LINKED ISSUE #722: Chrome Extension: Use latest rest request

### LINKED ISSUE #725: Chrome Extension: Pull only necessary fields

### LINKED ISSUE #726: Chrome Extension: Refactor

### PR DESCRIPTION
**Pull request by AdityaKhatri**
Fix https://github.com/the-deep/deeper/issues/726
Fix https://github.com/the-deep/deeper/issues/722
Fix https://github.com/the-deep/deeper/issues/725
Fix https://github.com/the-deep/deeper/issues/721

### CODE REVIEW

File: src/utils/faram.js [hunk 1]
```diff
@@ -5,7 +5,6 @@ export const alterResponseErrorToFaramError = (e) => {
         errors = e.reduce(
             (acc, error) => {
                 const {
-                    internalNonFieldErrors = [],
```
Comment (tnagorra) [inline src/utils/faram.js:8]: Why removed this line?

File: src/utils/faram.js [hunk 2]
```diff
@@ -34,7 +33,6 @@ export const alterResponseErrorToFaramError = (e) => {
     }
 
     const {
-        internalNonFieldErrors = [],
```
Comment (tnagorra) [inline src/utils/faram.js:37]: Why remove this line?

File: src/views/Navbar/index.js [hunk 3]
```diff
@@ -1,45 +1,31 @@
...
+Navbar.propTypes = {
+    title: PropTypes.string.isRequired,
     className: PropTypes.string,
     rightComponent: PropTypes.func,
-    title: PropTypes.string.isRequired,
 };
 
-const defaultProps = {
+Navbar.defaultProps = {
     className: '',
```
Comment (tnagorra) [inline src/views/Navbar/index.js]: Default value for `className` should be `undefined`

File: src/utils/common.js [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    }
+    if (isObject(data)) {
+        let newData = {};
+        forEach(data, (k, val) => {
+            const newEntry = sanitizeResponse(val);
+            // Comment from Bibek:
+            // This was:
+            // if (newEntry) {
+            // But it failed when an empty string or 0 was returned.
+            if (newEntry !== null && newEntry !== undefined) {
```
Comment (tnagorra) [inline src/utils/common.js]: Please use `isNotDefined` here

File: src/views/AddLead/index.js [hunk 5]
```diff
@@ -90,23 +100,121 @@ const sourceInputLabel = 'Publisher';
...
+            ),
+        },
+        onMount: ({
+            props: {
+                inputValues: { project } = {},
+                setLeadOptions,
+            },
+        }) => {
+            if (!project || project.length <= 0) {
+                setLeadOptions({ leadOptions: emptyObject });
```
Comment (tnagorra) [inline src/views/AddLead/index.js]: Let's not do this please sir kind sir

**Review by tnagorra — CHANGES_REQUESTED** [review]
Please fix before it can be merged.

Also, need to change the README.md to reflect appropriate changes.

File: src/utils/faram.js (hunk 1 above)
Comment (AdityaKhatri) [reply src/utils/faram.js:8]: Was not being used

File: src/utils/faram.js (hunk 2 above)
Comment (AdityaKhatri) [reply src/utils/faram.js:37]: was not being used

File: src/views/AddLead/index.js [hunk 6]
```diff
@@ -476,7 +616,7 @@ export default class AddLead extends React.PureComponent {
                     <div className={styles.actionButtons}>
                         <PrimaryButton
                             type="submit"
-                            disabled={pending}
+                            pending={pending}
```
Comment (tnagorra) [inline src/views/AddLead/index.js]: Khatri sir tapai le esto garna hudaina thiyo

File: src/views/AddLead/index.js [hunk 7]
```diff
@@ -430,12 +566,14 @@ export default class AddLead extends React.PureComponent {
...
                     pending={pending}
                 >
+                    <NonFieldErrors faramElement />
                     <MultiSelectInput
                         faramElementName="project"
                         label={projectInputLabel}
                         options={projects}
                         keySelector={projectKeySelector}
                         labelSelector={projectLabelSelector}
+                        hideClearAllButton
```
Comment (tnagorra) [inline src/views/AddLead/index.js]: I think we should have a clear all button for multi select

File: src/views/AddLead/index.js [hunk 8]
```diff
@@ -430,12 +566,14 @@ export default class AddLead extends React.PureComponent {
                     value={inputValues}
                     pending={pending}
```
Comment (tnagorra) [inline src/views/AddLead/index.js]: Do we have a pending in Faram?

File: src/views/AddLead/index.js [hunk 9]
```diff
@@ -127,84 +250,19 @@ export default class AddLead extends React.PureComponent {
                 ],
                 website: [requiredCondition],
             },
+            /*
```
Comment (tnagorra) [inline src/views/AddLead/index.js]: Line farming much?

File: src/utils/faram.js (hunk 1 above)
Comment (tnagorra) [reply src/utils/faram.js:8]: This was used to capture internalNonFieldErrors

File: src/utils/faram.js (hunk 2 above)
Comment (tnagorra) [reply src/utils/faram.js:37]: This was used to capture internalNonFieldErrors

**Pull request merged**